test/dom/dom.ts

```diff
@@ -0,0 +1,493 @@
+import {assert} from "chai";
+import JSBI from "jsbi";
+import * as ion from "../../src/Ion";
+import {IonTypes} from "../../src/Ion";
+import {Value} from "../../src/dom";
+import {encodeUtf8} from "../../src/IonUnicode";
+
+/**
+ * This file contains two tests for each Ion data type:
+ *   1.  'load() _______ as Value'
+ *       Demonstrates interacting with DOM nodes via the strongly-typed 'Value' interface.
+ *   2.  'load() _______ as any'
+ *       Demonstrates interacting with DOM nodes without type information (by casting them as 'any').
+ *
+ * A 'kitchen sink' example that loads a multi-value, heterogeneous Ion stream is also provided.
+ */
+
+describe('DOM', () => {
+   it('load() kitchen sink as Value[]', () => {
+      let values: Value[] = ion.loadAll(
+            ' 7' +
+            ' greeting::"Hello"' +
+            ' [\'moose\', null.string]' +
+            ' 5e1' +
+            ' {a: 2.1, b: 4, c: [foo, (bar baz), "qux"]}' +
+            ' 1970-01-01T00:00:00.000Z'
+      )!;
```

Comment:
is this `!` required?  `loadAll`'s signature doesn't specify `null` as a possible return;  maybe this is because we haven't enabled "strict" mode yet?

Reply:
It's required in `strict` mode, yes. I originally wrote this as a separate lib with `strict` enabled and got used to it.

`loadAll` returns an array, so if there aren't any values it just returns an empty array. `load` returns the first value, so if there isn't one it returns `null`. Based on the `null` vs `throws` elsewhere, it should probably `throw` instead.